src/main/java/frc/robot/poseestimator/PoseEstimationMath.java

```diff
@@ -0,0 +1,166 @@
+package frc.robot.poseestimator;
+
+import edu.wpi.first.math.geometry.*;
+import frc.robot.constants.Field;
+import frc.robot.poseestimator.observations.IRobotPoseVisionObservation;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class PoseEstimationMath {
+
+	public static Twist2d addGyroToTwist(Twist2d twist, Rotation2d currentGyroAngle, Rotation2d lastGyroAngle) {
+		boolean hasGyroUpdated = currentGyroAngle != null;
+		if (hasGyroUpdated && lastGyroAngle != null) {
```

Comment:
the meaning of the hasGyroUpdated is not clear, and why isn't there another one for the lastGyroAngle?